Source/JavaScriptCore/API/glib/JSCValue.cpp

```diff
@@ -2145,3 +2145,30 @@ char* jsc_value_to_json(JSCValue* value, unsigned indent)
 
     return json;
 }
+
+/**
+ * jsc_value_new_deferred_promise:
+ * @context: a #JSCContext
+ * @value: a #JSCValue
+ *
+ * Create a new Promise returning handles to resolve and reject method
+ *
+ * Returns: (transfer full): a defered promise
```

Comment:
Small typo, and it would be good to indicate the returned value is an object:

```suggestion
 * Returns: (transfer full): a deferred promise object
```

Adding a small code example to the documentation about how to use a promise object created with this API would be nice, too.

Reply:
I agree, by documentation you mean here in the comment section or you have a specific file in mind ?